.github/workflows/ci.yml

```
@@ -1,6 +1,10 @@
 name: Continuous integration
 on: [push, pull_request]
 
+concurrency:
+  group: ${{ github.workflow}}-${{ github.ref }}
```

just a styling nitpick

```suggestion
  group: ${{ github.workflow }}-${{ github.ref }}
```